apps/src/applab/DesignToggleRow.jsx

```diff
@@ -47,9 +49,43 @@ module.exports = React.createClass({
   render: function () {
     var selectDropdown;
     var dropdownStyle = {
-      width: 140,
-      marginLeft: 10
+      display: 'inline-block',
+      verticalAlign: 'top',
+      width: 130,
+      height: 28,
+      marginBottom: 6,
+      borderColor: '#949ca2'
+    };
+
+    var buttonStyle = {
+      display: 'inline-block',
+      verticalAlign: 'top',
+      border: '1px solid #949ca2',
+      margin: '0 0 8px 0',
+      padding: '3px 6px',
+      fontSize: 14
+    };
+    var buttonPrimary = {
+      backgroundColor: '#ffa000',
+      color: '#fff'
     };
+    var buttonSecondary = {
+      backgroundColor: '#e7e8ea',
+      color: '#949ca2'
+    };
+
+    var codeButtonStyle = $.extend({}, buttonStyle, {
+      borderBottomRightRadius: 0,
+      borderTopRightRadius: 0,
+      borderRightWidth: 0
+    }, (this.state.mode === Mode.CODE ? buttonSecondary : buttonPrimary));
+
+    var designButtonStyle = $.extend({}, buttonStyle, {
+      borderBottomLeftRadius: 0,
+      borderTopLeftRadius: 0
+    }, (this.state.mode === Mode.CODE ? buttonPrimary : buttonSecondary));
```

Comment:
Both style libraries I mentioned actually give shoutouts to that particular talk. I think they would make it possible to do things like :focus, etc. inline as well.

I think what you have is fine. Mostly just throwing I was just throwing this out there as something to think about if things continue to expand in complexity, as it could make things easier.